articles/5_sass_functions.md

```diff
@@ -0,0 +1,337 @@
+
+### What is a `z-index`?
+
+The `z-index` of an element refers to its positioning in the z-axis. If you think of the x- and y- axes as the left-right / up-down directions on the screen, the z-axis is the direction going into and out of the screen. Another way of thinking about it is the way the element layers above or below other elements on the page.
+
+![z-index](../images/z-index.gif)
+
+*Image courtesy of http://www.annyhe.com/*
+
+Something to note that is not directly related to the larger topic of this post, but is related to `z-index`es: When applying a `z-index` to elements nested within each other, the child element's maximum place in the layering order is that of its parent, no matter how high the child element's `z-index` is.
+
+For a visual representation of how that looks (Assume that boxes 2 and 3 are children of box 1):
+
+![stacking order](../images/stacking1.png)
+
+*Image courtesy of webdesign.tutsplus.com*
+
+For this reason, its important to avoid putting `z-index`es on elements that don't absolutely need them, as it can lead to some unintended consequences.
+
+### Why did I need to refactor them?
+
+Initially, all of our `z-index`es were stored as individual variables in a file that looked like this:
+
+```scss
+$zindex-navbar:                  1000;		
+$zindex-dropdown:                1000;		
+$zindex-popover:                 1010;		
+$zindex-tooltip:                 1030;		
+$zindex-navbar-fixed:            1030;		
+$zindex-modal-background:        1040;		
+$zindex-modal:                   1050;
+...
+```
+
+While these initial values are fairly well organized, this quickly fell into chaos as every new component or one-off change to a `z-index` just led to creating new variables with magic numbers, or worse, just putting the magic number directly into the declaration without explanation for why that element had to be layered that way.
+
+### How did I fix it?
+
+Now, our `z-index`es are organized in a nested map structure based on where they appear in the app.
+
+```scss
+$z-layers: (
+  'base': 0,
+  'input': 1,
+  'tool-container': 3,
+  'loading-animation': 999,
+  'navbar': 1000,
+  'modal': 1050,
+  'time-series': 1060,
+  'customer-support': 1070,
+  'popover': 1080,
+  'tooltip': 1090,
+  'alert': 2000,
+  'context-chooser': (
+    'base': 2,
+    'collapse': 4
+  ),
+  'icons': (
+    'caret': 2,
+    'sort': 2,
+    'recon-expand-table': 100
+  ),
+  'portal': (
+    'header': 3,
+    'nav': 10
+  ),
+  'report': (
+    'widgets': 2,
+    'footer': 10,
+    'container': 100,
+    'drop-zone': 110,
+    'header-cover': 400,
+    'nav': 500,
+    'item-outline': 1000,
+    'text-overflow': 1120,
+    'page': (
+      'linked-cue': 500
+    ),
+    'table': (
+      'squash-column': 3
+    )
+  )
+);
+```
+
+I also created this function to access the values. This function takes any number of arguments, and traverses the structure by calling `map-get` and `map-deep-get`, an internal function outlined below.
```

Comment:
I think it'd be nice to have a longer description of what this function does exactly.